trunk/src/app/srs_app_server.cpp

```diff
@@ -1378,6 +1378,8 @@ srs_error_t SrsServer::accept_client(SrsListenerType type, srs_netfd_t stfd)
             srs_close_stfd(stfd); srs_error_reset(err);
             return srs_success;
         }
+        //close fd on conn error, otherwise will lead to fd leak -gs
+        srs_close_stfd(stfd);
```

Comment:
看起来，比较好的做法，就是在错误了之后全部关闭stfd，而不用关闭两次。

```
    
    if ((err = fd_to_resource(type, stfd, &conn)) != srs_success) {
        //close fd on conn error, otherwise will lead to fd leak -gs
        srs_close_stfd(stfd);
        if (srs_error_code(err) == ERROR_SOCKET_GET_PEER_IP && _srs_config->empty_ip_ok()) {
            srs_error_reset(err);
            return srs_success;
        }

Reply:
嗯 是的 我改一下子

Reply:
已更新